adds internal monologue attack module

audibleblink opened the pull request:
This PR pulls in changes made to `master` since the last divergence of `dev` from `master` (these would have surfaced merge conflicts on the next merge from dev into master)

Then it adds a module for Elad Shamir's Internal Monologue attack. - https://github.com/eladshamir/Internal-Monologue/

xorrior committed:
Fix macho generation logic

xorrior committed:
Agent display improvements

Ret2lib committed:
fix keylogger for newer ruby + stderr redirection

This will add support for fiddle which is recommended instead of 'DL'. Versions 2.3 and above will use Fiddle, while versions before that will use DL. This will log stderr to logfile as well instead of outputing to terminal (better opsec)

stempelo committed:
Update README.md

Change workspace Slack

xorrior committed:
Merge pull request #1168 from stempelo/patch-1

Update README.md

audibleblink committed:
adds internal monologue module

ThePirateWhoSmellsOfSunflowers commented on lib/modules/powershell/credentials/invoke_internal_monologue.py:
```diff
@@ -0,0 +1,118 @@
+
+            'NeedsAdmin': True,
+
+            'OpsecSafe': True,
```
OpsecSafe is not guaranteed if the downgrade to NetNTLMv1 is performed. You should switch this value.

audibleblink replied:
I can see that. What would you consider a better approach?

audibleblink replied:
I can see that. What would you consider a better approach?

* Switch Opesec to False
* Disable Impersonation and the downgrade as default values

ThePirateWhoSmellsOfSunflowers replied:
Both solutions are the same, so there is no a better approach in my opinion. Maybe you can switch Opsec to False but add in the description of Impersonation/Downgrade something like "If disabled then the opsec may be better".

StayPirate committed:
Update Slack link

n0pe-sled committed:
Launch Agent Userland Persistence

xorrior committed:
Merge pull request #1190 from TuxMeaLux/patch-1

Update Slack link

ktlmsney committed:
Fix urllib3

audibleblink commented:
Anyone able to test this yet?

awsmhacks committed:
Update/Fix Invoke-Kerberoast

Commenting out, though it looks like it could be removed, two sections that throw errors. New-DynmaicParameter is not a function and I couldn't find it anywhere besides helper scripts on technet. 
Looks like it was first introduced in https://github.com/EmpireProject/Empire/commit/71c795a4489b133cfad7d599f9d92a1004db5934#diff-494cc831008b28a810840a12d052d51e

rvrsh3ll committed:
Merge pull request #1357 from awsmhacks/patch-1

Update/Fix Invoke-Kerberoast

xorrior committed:
Update README.md

vinnybod committed:
Hubble's BCSec changes

vinnybod committed:
remove temp stager file

vinnybod committed:
Merge pull request #1 from BC-SECURITY/hubble-changes-2

Hubble changes

vinnybod committed:
Merge branch 'master' into dev

vinnybod committed:
Merge pull request #2 from BC-SECURITY/dev

Dev

Hubbl3 committed:
python 3 stage one updates

Hubbl3 committed:
python 3 stage 2/intial running config updates

Hubbl3 committed:
python 3 first round of py 2 remediation

Hubbl3 committed:
Http listener with Multi/launcher working

vinnybod committed:
remove venv

Hubbl3 committed:
fix for issue #5

Commit:
Updated empire.py and requirements.txt

Cx01N committed:
Fix for port issue - issue #11

Cx01N committed:
Fixed PSinject - issue #6

Commit:
python3 multi/launcher working. Lots of debug statements printing

Cx01N committed:
Fixed agent tasking

Cx01N committed:
Updated PSinject compatibility with Python 3

Hubbl3 committed:
converted bytes wrappers

Hubbl3 committed:
intial beta of 2/3 crossover functioning

Cx01N committed:
Update README.md

Cx01N committed:
Fixed Mimikatz error #15

Cx01N committed:
Fixed Python 3 tab complete - Issue #13)

Cx01N committed:
Http_com update

Cx01N committed:
Fixed listener http_com

Cx01N committed:
Update Mimikatz/dcsync

Cx01N committed:
Cleaned up debug messages

Cx01N committed:
Fixed http_hop token issue #17
